egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,352 @@
+
+
+# This function replicate the entries in files like segments, utt2spk, text
+def AddPrefixToFields(input_file, output_file, num_replicas, prefix, field = [0]):
+    list = map(lambda x: x.strip(), open(input_file))
+    f = open(output_file, "w")
+    for i in range(num_replicas):
+        for line in list:
+            if len(line) > 0 and line[0] != ';':
+                split1 = line.split()
+                for j in field:
+                    if prefix is not None:
+                        split1[j] = prefix + str(i) + "_" + split1[j]
+                print(" ".join(split1), file=f)
+            else:
+                print(line, file=f)
+    f.close()
+
+
+def CreateReverberatedCopy(input_dir, output_dir, room_dict, noise_list, foreground_snr_string, background_snr_string, num_replicas, prefix, speech_rvb_probability, noise_adding_probability, max_noises_per_minute):
+    
+    if not os.path.isfile(input_dir + "/reco2dur"):
+        print("Getting the duration of the recordings...");
+        train_lib.RunKaldiCommand("wav-to-duration --read-entire-file=true scp:{0}/wav.scp ark,t:{0}/reco2dur".format(input_dir))
+    durations = ParseFileToDict(input_dir + "/reco2dur", value_processor = lambda x: float(x[0]))
+    wav_scp = ParseFileToDict(input_dir + "/wav.scp", value_processor = lambda x: " ".join(x))
+    foreground_snr_array = map(lambda x: float(x), foreground_snr_string.split(':'))
+    background_snr_array = map(lambda x: float(x), background_snr_string.split(':'))
+
+    CorruptWav(wav_scp, durations, output_dir, room_dict, noise_list, foreground_snr_array, background_snr_array, num_replicas, prefix, speech_rvb_probability, noise_adding_probability, max_noises_per_minute)
+
+    AddPrefixToFields(input_dir + "/utt2spk", output_dir + "/utt2spk", num_replicas, prefix, field = [0,1])
+    train_lib.RunKaldiCommand("utils/utt2spk_to_spk2utt.pl <{output_dir}/utt2spk >{output_dir}/spk2utt"
+                    .format(output_dir = output_dir))
+
+    if os.path.isfile(input_dir + "/text"):
+        AddPrefixToFields(input_dir + "/text", output_dir + "/text", num_replicas, prefix, field =[0])
+    if os.path.isfile(input_dir + "/segments"):
+        AddPrefixToFields(input_dir + "/segments", output_dir + "/segments", num_replicas, prefix, field = [0,1])
+    if os.path.isfile(input_dir + "/reco2file_and_channel"):
+        AddPrefixToFields(input_dir + "/reco2file_and_channel", output_dir + "/reco2file_and_channel", num_replicas, prefix, field = [0,1])
+
+    train_lib.RunKaldiCommand("utils/validate_data_dir.sh --no-feats {output_dir}"
+                    .format(output_dir = output_dir))
+
+def SmoothProbability(list):
+    uniform_probability = 1 / float(len(list))
+    for item in list:
+        if item.probability is None:
+            item.probability = uniform_probability
+        else:
+            # smooth the probability
+            item.probability = 0.3 * item.probability + 0.7 * uniform_probability
```

Comment:
@vijayaditya I remember we both agree on using 0.7 for the weight of the uniform_probability. This value decides how heavy you want the smoothing to be